components/server/src/websocket-connection-manager.ts

```diff
@@ -67,6 +70,7 @@ export class WebsocketConnectionManager<C extends GitpodClient, S extends Gitpod
             resourceGuard = new CompositeResourceAccessGuard([
                 new OwnerResourceGuard(user.id),
                 new SharedWorkspaceAccessGuard(),
+                new WorkspaceLogAccessGuard(() => gitpodServer.getLoggedInUser(), this.hostContextProvider),
```

Comment:
```suggestion
                new WorkspaceLogAccessGuard(() => user, this.hostContextProvider),
```

much like in the `OwnerResourceGuard` above

Reply:
Hm, would be good to quickly talk about this. IMO it would be best to avoid the `User` (including possibly outdated state!) here and just use the `id`, so clients are forced to do the `User` lookup (and thus get up-to-date data).

Reply:
Especially regarding `blocked` it _seems_ this is orthogonal (until now), but it would also make sense to disallow log access in blocked state.
If this is a concern of the `WorkspaceLogAccessGuard`, I'd require an "up-to-date" `User`, and not the one from when I established the websocket connection.

Reply:
We discussed offline that `blocked` is an orthogonal issue.